### PR TITLE
Minimal 9slice border-image support

### PR DESCRIPTION
**Pull request by ogoffart**
- Doesn't support tiling yet, only stretching
 - Added a `9slice(...)` attribute to the @image-url
 - Added API in C++ and Rust to set the edge value.  (JS is missing)
 - Added rendering support in all our renderer

CC https://github.com/slint-ui/slint/issues/1529

### CODE REVIEW

File: api/cpp/cbindgen.rs [hunk 1]
```diff
@@ -463,14 +464,15 @@ fn gen_corelib(
...
+                "slint_image_set_nine_slice_edges",
                 "SharedPixelBuffer",
                 "SharedImageBuffer",
                 "StaticTextures",
                 "BorrowedOpenGLTextureOrigin"
             ],
             vec!["Color"],
             "slint_image_internal.h",
-            "namespace slint::cbindgen_private { struct ParsedSVG{}; struct HTMLImage{}; using namespace vtable; }",
+            "namespace slint::cbindgen_private { struct ParsedSVG{}; struct HTMLImage{}; using namespace vtable; namespace types{struct NineSliceImage{}; } }",
```
Comment (tronical) [inline api/cpp/cbindgen.rs:475]:
```suggestion
            "namespace slint::cbindgen_private { struct ParsedSVG{}; struct HTMLImage{}; using namespace vtable; namespace types{ struct NineSliceImage{}; } }",
```

File: docs/reference/src/language/builtins/elements.md [hunk 2]
```diff
@@ -298,12 +298,13 @@ An `Image` can be used to represent an image loaded from a file.
 
 -   **`colorize`** (_in_ _brush_): When set, the image is used as an alpha mask and is drawn in the given color (or with the gradient).
 -   **`horizontal-alignment`** (_in_ _enum [`ImageHorizontalAlignment`](enums.md#imagehorizontalalignment)_): The horizontal alignment of the image within the element.
--   **`image-fit`** (_in_ _enum [`ImageFit`](enums.md#imagefit)_): Specifies how the source image shall be fit into the image element. (default value: `contain` when the `Image` element is part of a layout, `fill` otherwise)
+-   **`image-fit`** (_in_ _enum [`ImageFit`](enums.md#imagefit)_): Specifies how the source image shall be fit into the image element. Do not have effect with 9 slice scaling images.
```
Comment (tronical) [inline docs/reference/src/language/builtins/elements.md:301]:
```suggestion
-   **`image-fit`** (_in_ _enum [`ImageFit`](enums.md#imagefit)_): Specifies how the source image shall be fit into the image element. Does not have any effect when used with 9 slice scaling images.
```

File: docs/reference/src/language/builtins/elements.md [hunk 3]
```diff
@@ -353,6 +354,20 @@ export component Example inherits Window {
 }
 ```
 
+Esample using 9slice
```
Comment (tronical) [inline docs/reference/src/language/builtins/elements.md:357]:
```suggestion
Example using 9slice:
```

File: internal/backends/qt/qt_window.rs [hunk 4]
```diff
@@ -1253,40 +1253,53 @@ impl QtItemRenderer<'_> {
...
-                pixmap as "QPixmap",
-                source_rect as "QRectF",
-                dest_rect as "QRectF",
-                smooth as "bool"] {
-            (*painter)->save();
-            (*painter)->setRenderHint(QPainter::SmoothPixmapTransform, smooth);
-            (*painter)->drawPixmap(dest_rect, pixmap, source_rect);
-            (*painter)->restore();
-        }};
+        for fit in fit {
```
Comment (tronical) [inline internal/backends/qt/qt_window.rs:1275]: This makes me think that the vector should be called “fits” or “fitted_images”.

File: internal/compiler/parser/expressions.rs [hunk 5]
```diff
@@ -417,3 +412,80 @@ fn parse_tr(p: &mut impl Parser) {
     }
     p.expect(SyntaxKind::RParent);
 }
+
+/// ```test,AtTr
```
Comment (tronical) [inline internal/compiler/parser/expressions.rs:416]:
```suggestion
/// ```test,AtImageUrl
```

File: tests/cases/elements/image.slint [hunk 6]
```diff
@@ -13,6 +13,8 @@ TestCase := Rectangle {
         source-clip-x: 20;
     }
 
+    out property <image> with-border: @image-url("dog.jpg", 9slice(12 13 14 15));
```
Comment (tronical) [inline tests/cases/elements/image.slint:16]: This tests basically the code generators, right?

**Review by tronical — APPROVED** [review]

File: internal/compiler/parser/expressions.rs (hunk 5 above)
Comment (ogoffart) [reply internal/compiler/parser/expressions.rs:416]: I also forgot the `#[cfg_attr(test, parser_test)]` which is why it did not fail

```suggestion
#[cfg_attr(test, parser_test)]
/// ```test,AtImageUrl
```

File: tests/cases/elements/image.slint (hunk 6 above)
Comment (ogoffart) [reply tests/cases/elements/image.slint:16]: Yes. We do not have test for renderers other than the software renderer, and that's only tested with rust.

**Comment by ogoffart** [comment]
@tronical I'll merge this soon if you don't see an objection.

**Review by tronical — APPROVED** [review]
Not at all - looks good to me:)

**Pull request merged**